src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -721,26 +721,24 @@ public boolean equals(Object other) {
             if (this == other) {
                 return true;
             }
-            try {
-                final CacheKey otherEntry = (CacheKey)other;
+            if (other instanceof CacheKey otherEntry) {
                 //quick check to see if they are not equal
                 if (modulesHash != otherEntry.modulesHash) {
                     return false;
                 }
                 //are the names the same?
-                if (!name.equals(otherEntry.name)) {
+                if (!Objects.equals(name, otherEntry.name)) {
                     return false;
                 }
                 // are the locales the same?
-                if (!locale.equals(otherEntry.locale)) {
+                if (!Objects.equals(locale, otherEntry.locale)) {
                     return false;
                 }
                 // are modules and callerModules the same and non-null?
                 Module module = getModule();
                 Module caller = getCallerModule();
                 return ((module != null) && (module.equals(otherEntry.getModule())) &&
                         (caller != null) && (caller.equals(otherEntry.getCallerModule())));
-            } catch (NullPointerException | ClassCastException e) {
```

Comment:
Are we sure that NPE can only be thrown when calling equals on `name` or `locale` and not, for example, from `getModule()` or `getCallerModule()`?

Reply:
Yes:
```java
CacheKey(String baseName, Locale locale, Module module, Module caller) {
    Objects.requireNonNull(module);
    Objects.requireNonNull(caller);

    this.name = baseName;
    this.locale = locale;
    this.moduleRef = new KeyElementReference<>(module, referenceQueue, this);
    this.callerRef = new KeyElementReference<>(caller, referenceQueue, this);
    this.modulesHash = module.hashCode() ^ caller.hashCode();
}

Reply:
I'll run our CI, and if all good, I'll approve this PR. If nothing else, this change seems reasonable and correct.

Reply:
Thanks Pavel for jumping in. I too think this change looks good.

Reply:
Yep; all good.